pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/SourcesImpl.java

```diff
@@ -741,6 +742,7 @@ public SourceConfig getSourceInfo(final String tenant,
         SourceConfigUtils.ExtractedSourceDetails sourceDetails
                 = SourceConfigUtils.validateAndExtractDetails(
                         sourceConfig, classLoader, worker().getWorkerConfig().getValidateConnectorConfig());
+        ClassLoaderUtils.closeClassLoaderIfNeed(classLoader);
```

Comment:
I wonder in which case the `classLoader` is instance of `Closeable`

Reply:
UrlClassLoader is instance of `Closable`. And we are using NarClassLoader widely in pulsar-functions, which is the subClass of UrlClassLoader

Reply:
Ok, good job.